src/ConservationChecker.jl

```diff
@@ -0,0 +1,331 @@
+        @. cc.ρe_tot_atmos + cc.ρe_tot_ocean + cc.ρe_tot_land + cc.ρe_tot_seaice + cc.toa_net_source + cc.friction_sink
+    # @assert abs((tot[end] - tot[1]) / tot[end]) < 1e-4
+end
+
+"""
+    ke_dissipation(sim)
+
+Calculate the kinetic energy diffipation of a simulation. 
+"""
+function ke_dissipation(sim)
+    drag_uv =
+        .-Geometry.UVVector.(
+            Geometry.Covariant12Vector.(
+                sim.integrator.p.dif_flux_uₕ_bc.components.data.:1,
+                sim.integrator.p.dif_flux_uₕ_bc.components.data.:2,
+            )
+        )
+    dot.(Geometry.UVVector.(level(sim.integrator.u.c.uₕ, 1)), drag_uv) .* level(sim.integrator.u.c.ρ, 1)
+end
+
+"""
+        check_conservation(
+        cc::WaterConservationCheck,
+        coupler_sim,
+        )
+
+computes the total water ∫ ρq_tot dV of the various components
+of the coupled simulations, and updates cc with the values.
+
+Note: in the future this should not use `push!`.
+"""
+function check_conservation!(
+    cc::WaterConservationCheck,
+    coupler_sim::CoupledSimulation,
+    get_slab_energy,
+    get_land_energy,
+)
+    @unpack model_sims, surface_masks = coupler_sim
+    @unpack atmos_sim, land_sim, ocean_sim, ice_sim = model_sims
+
+    boundary_space = coupler_sim.boundary_space
+    FT = eltype(coupler_sim.surface_masks.land)
+
+    # save atmos
+    push!(cc.ρq_tot_atmos, sum(atmos_sim.integrator.u.c.ρq_tot))
+
+    # save land
+    if land_sim !== nothing
+        water_content =
+            @. (land_sim.integrator.u.bucket.σS + land_sim.integrator.u.bucket.W + land_sim.integrator.u.bucket.Ws) # m^3 water / land area / layer height
+        parent(water_content) .= parent(water_content .* surface_masks.land) * FT(1000)
+        push!(cc.ρq_tot_land, sum(water_content))
+    else
+        push!(cc.ρq_tot_land, FT(0))
+    end
+
+    # save sea ice
+    coupler_sim.fields.P_net .-= swap_space!(surface_water_gain_from_rates(coupler_sim), boundary_space) # accumulated surface water gain 
+    if ice_sim !== nothing
+        push!(cc.ρq_tot_seaice, sum(coupler_sim.fields.P_net .* surface_masks.ice))
+    else
+        push!(cc.ρq_tot_seaice, FT(0))
+    end
+
+    # save ocean
+    if ice_sim !== nothing
+        push!(cc.ρq_tot_ocean, sum(coupler_sim.fields.P_net .* surface_masks.ocean))
+    else
+        push!(cc.ρq_tot_ocean, FT(0))
+    end
+
+    tot = @. cc.ρq_tot_atmos + cc.ρq_tot_land + cc.ρq_tot_seaice + cc.ρq_tot_ocean
+    # @assert abs((tot[end] - tot[1]) / tot[end]) < 1e-2
+end
+
+"""
+    surface_water_gain_from_rates(cs)
+
+Determines the total water content gain/loss of a surface from the begining of the simulation based on evaporation and precipitation rates. 
+"""
+function surface_water_gain_from_rates(cs::CoupledSimulation)
+    evaporation = cs.fields.F_E # kg/m^2/s / layer depth
+    precipitation_l = cs.fields.P_liq
+    precipitation_s = cs.fields.P_snow
+    @. (evaporation + precipitation_l + precipitation_s) * cs.Δt_cpl # in kg 
```

Comment:
```suggestion
    @. (evaporation + precipitation_l + precipitation_s) * cs.Δt_cpl # in kg /m^2
```
I think that's the unit? I'm not sure though.

Reply:
see above. Now clarified